website/docs/gen-ai/inference/rayserve-ha.md

```diff
@@ -0,0 +1,208 @@
+
+GCS fault tolerance requires an external Redis database. You can choose to host your own Redis database, or you can use one through a third-party vendor.
+
+For development and testing purposes, you can also host a containerized Redis database on the same EKS cluster as your Ray cluster. However, for production setups, it's recommended to use a highly available external Redis cluster. In this pattern, we've used [Amazon ElasticCache for Redis](https://aws.amazon.com/elasticache/redis/) to create an external Redis cluster. You can also choose to use [Amazon memoryDB](https://aws.amazon.com/memorydb/) for setting up a Redis cluster.
+
+As part of the current blueprint, we've added a terraform module named `elasticache` that creates an Elastic Cache Redis cluster in AWS. This uses The Redis cluster has cluster mode disabled and contain one node. This cluster node's endpoint can be used for both reads and writes.
+
+Key things to note in this module are -
+
+- The Redis Cluster is in the same VPC as the EKS cluster. If the Redis cluster is created in a separate VPC, then VPC peering needs to be set up between the EKS cluster VPC and the Elastic Cache Redis cluster VPC to enable network connectivity.
+- A cache subnet group needs to be created at the time of creating the Redis cluster. A subnet group is a collection of subnets that you may want to designate for your caches in a VPC. ElastiCache uses that cache subnet group to assign IP addresses within that subnet to each cache node in the cache. The blueprint automatically adds all the subnets used by the EKS cluster in the subnet group for the Elastic cache Redis cluster.
+- Security Group - The Security Group assigned to the Redis cache needs to have an inbound rule that allows TCP traffic from EKS Cluster's worker node security group to the Redis cluster security group over port 6379. This is because the Ray head Pod needs to establish a connection to the Elastic cache Redis cluster over port 6379. The blueprint automatically sets up the security group with the inbound rule.
+
+To create the Redis cluster using Amazon Elastic Cache, please follow the below steps.
+
+:::info
+
+This Mistral7b deployment is using Ray Serve with High availability. If you have already deployed mistral7b in the previous steps, then you can delete the deployment and run the below steps.
+
+:::
+
+**Prerequisites**:
+
+1. [aws cli](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html)
+2. [kubectl](https://Kubernetes.io/docs/tasks/tools/)
+3. [terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+4. [envsubst](https://pypi.org/project/envsubst/)
+5. [jq](https://jqlang.github.io/jq/download/)
+
+First, enable the creation of the Redis cluster by setting the `enable_rayserve_ha_elastic_cache_redis` variable to `true` in `variables.tf` file. By default it's set to `false`.
+
+Then, run the `install.sh` script to install the EKS cluster with KubeRay operator and other add ons.
+
+```bash
```

Comment:
add a step to enable `export TF_VAR_enable_rayserve_ha_elastic_cache_redis=true`